main.tf

```diff
@@ -96,3 +156,6 @@ resource "aws_s3_bucket" "tf_backend_logs_bucket" {
   }
 }
 
+data "aws_kms_key" "master" {
+  key_id = "alias/aws/s3"
```

Comment:
I wonder if you could/should do this?

```
data "aws_kms_key" "master" {
  key_id = "${var.kms_key_id == "" ? "alias/aws/s3" : var.kms_key_id}"
}
```
This would demand its existence on the tf run though.........

But you could modify your output variable to a static reference to this data output.

I am not sure if there are any other implications here. Let me know what your thoughts are on this.

Reply:
this seems cleaner, i like it